src/Xamarin.Android.Build.Tasks/Tests/Xamarin.Android.Build.Tests/SingleProjectTest.cs

```diff
@@ -52,6 +54,8 @@ public void AndroidManifestProperties (string versionName, string versionCode, s
 				proj.SetProperty ("GenerateApplicationManifest", "true");
 			}
 			proj.SetProperty ("ApplicationId", packageName);
+			proj.SetProperty ("ApplicationIcon", applicationIcon);
```

Comment:
The implied semantics here are *weird* to me.

The primary idea is (1) use in MAUI Single Project, and (2) the implied usage would be a `.csproj` with:

```xml
<PropertyGroup>
  <AndroidIcon>@drawable/icon</AndroidIcon>
  <AndroidRoundIcon>@drawable/icon</AndroidRoundIcon>
</PropertyGroup>
```

How is this going to work on iOS?  WinUI?  Tizen?  Shouldn't these properties instead refer to *files*?

```xml
<PropertyGroup>
  <AndroidIcon>Resources\Icon.svg</AndroidIcon>
  <AndroidRoundIcon>Resources\RoundIcon.svg</AndroidRoundIcon>
</PropertyGroup>
```

Reply:
This does bring up the qusetion around `@mipmap` vs `@drawable` as both are valid, which one will Maui generate?

Reply:
The MAUI example above is:
```xml
<MauiIcon Include="Resources\AppIcon\appicon.svg" ForegroundFile="Resources\AppIcon\appiconfg.svg" Color="#512BD4" />
```
The "Resizetizer" MSBuild logic currently creates mipmaps:

![image](https://user-images.githubusercontent.com/840039/193049187-a80c9ddd-4cc1-4bda-b7f7-5bad7a692ded.png)

There is some missing design here though...

Customers in a plain .NET 6 Android project can't set a file path, because an app icon is multiple files:

    Resources/mipmap-hdpi/icon.png
    Resources/mipmap-mdpi/icon.png
    Resources/mipmap-xhdpi/icon.png
    .etc.

So then how useful is `ApplicationIcon=@mipmap/icon`? @mattleibow are the properties above sufficient for what MAUI needs?

Reply:
Should we be using just the resource name instead? and have a task which will look for the right files in @mipmap then @drawable ? e.g `ApplicationIcon=icon`.  So it will then just look for a resource called `icon.png`...

Maybe get other platforms to do the same?

If we have a `@xxx` at the start, maybe skip the searchng part , at least for android.

Reply:
This thread doesn't quite answer my question, and I imagine we need @mattleibow's input, so, to rephrase:

Given the [maui-mobile](https://github.com/dotnet/maui/blob/main/src/Templates/src/templates/maui-mobile/MauiApp.1.csproj) template, what changes do we want to have present?  What makes sense in a multi-UI world?

Having `$(ApplicationIcon)=@drawable/icon` does not make sense to me, as I do not understand how that would make sense to iOS or Tizen or WinUI or…

Having `$(ApplicationIcon)` match a `@(MauiIcon)` value *does* make sense to me, as `@(MauiIcon)` applies to all UI, but I am still uncertain about that.  Can there be more than one `@(MauiIcon)`?  Can `%(MauiIcon.Include)` be duplicated?

Perhaps the "better" idea is, instead of an `$(ApplicationIcon)` property, we have `%(MauiIcon.ApplicationIcon)` metadata?

```xml
<ItemGroup>
  <MauiIcon
      Include="Resources\AppIcon\appicon.svg"
      ForegroundFile="Resources\AppIcon\appiconfg.svg"
      Color="#512BD4"
      ApplicationIcon="True"
  />
</ItemGroup>
```

…though that would permit "multiple" `%(MauiIcon.ApplicationIcon)`=True values, which is likely "Bad".

I'm just fundamentally not understanding the high-level usage and expectations.  https://github.com/dotnet/maui/pull/8063 doesn't help me, nor does https://github.com/xamarin/xamarin-android/issues/7147.

Then there's the [`@(MauiIcon)` documentation](https://learn.microsoft.com/en-us/dotnet/maui/user-interface/images/app-icons?tabs=android), which feels like whoever came up with this idea has no idea about MSBuild conventions.  If there can be only one, it *should not* have been an Item!  It should have been a Property!  (Except then you couldn't easily add metadata to the property, so that's weird…)  It should also *warn* if there is more than one and some will be ignored!  Just…bizarre.

Given the existence of `@(MauiIcon)`, does a cross-UI `$(ApplicationIcon)` need to exist?  Or should it instead be `$(AndroidApplicationIcon)`, which is derived from the `@(MauiIcon)` value?

Reply:
I just had a thought. Perhaps we need the `AndroidApplicationIcon` property (and iOS needs its one to) which we will use if present to set the manifest. 

Then Maui could add a task.property which is cross platform via meta data or a new property `MauiApplicationIcon` which will then generate the right file for the platform and then set the new `AndroidApplicationIcon` in android. It will do the same for other platforms.